### PR TITLE
cdc check healthy when rolling upgrade

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What is changed and how it works?

* introduce cdc healthy open api, check for cdc cluster healthy before perform rolling upgrade.
* enable ticdc rolling upgrade since `v6.3.0`, which is not public yet.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

Side effects

Related changes

Release notes:

```release-note
TiUP support TiCDC graceful rolling upgrade since 6.3.0
```

### CODE REVIEW

File: pkg/cluster/operation/operation.go [hunk 1]
```diff
@@ -27,6 +28,24 @@ const (
 	EnvNamePDEndpointOverwrite  = "FORCE_PD_ENDPOINTS"
 )
 
+type contextKey string
+
+// ContextKeyCurrentVersion is the key of current cluster version
+const ContextKeyCurrentVersion contextKey = "currentVersion"
+
+func getCurrentVersionFromContext(ctx context.Context) string {
```
Comment (nexustar) [inline pkg/cluster/operation/operation.go:36]: I suggest just put cluster version on args of upgrade function. There is no need to put it to context. rest LGTM

Comment (3AceShowHand) [reply pkg/cluster/operation/operation.go:36]: I've tried it, but it causes the signature to change, which makes a lot of other code changes.
 If it's acceptable, it's ok to me.

Comment (nexustar) [reply pkg/cluster/operation/operation.go:36]: I think it is acceptable, since we never promise api stable in tiup v1.x

Comment (3AceShowHand) [reply pkg/cluster/operation/operation.go]: resolved

**Review by nexustar — APPROVED** [review]

**Review by srstack — APPROVED** [review]

**Pull request merged**